zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -984,11 +997,50 @@ private void fireAllEvents() {
 
   protected List<String> getChildren(final String path, final boolean watch) {
     long startT = System.currentTimeMillis();
+
     try {
       List<String> children = retryUntilConnected(new Callable<List<String>>() {
+        private int connectionLossRetryCount = 0;
+
         @Override
         public List<String> call() throws Exception {
-          return getConnection().getChildren(path, watch);
+          try {
+            return getConnection().getChildren(path, watch);
+          } catch (ConnectionLossException e) {
+            ++connectionLossRetryCount;
```

Comment:
Do we need this? If it is connection loss, @jiajunwang 's improvement for retry at zk event thread layer can handle that, right?

Reply:
@dasahcc My purpose is to avoid stat check for each connection loss, because large children is just a very tiny possibility. And high possibility for network issue/session expiry. Then we could avoid the extra stat call in most of time.

// Allow retrying 3 times before checking stat checking number of
//  because there is a higher possibility that connection loss is caused by other
// factors such as network connectivity, connected ZK node could not serve
// the request, session expired, etc.